ext/data/schemas/options-schema.json

```diff
@@ -813,6 +783,39 @@
                                     }
                                 }
                             },
+                            "languages": {
+                                "type": "object",
+                                "propertyNames": {
+                                    "type": "string"
+                                },
+                                "additionalProperties": {
+                                    "type": "object",
+                                    "properties": {
+                                        "textTransformations": {
+                                            "type": "object",
+                                            "propertyNames": {
+                                                "type": "string"
+                                            },
+                                            "additionalProperties": {
+                                                "type": "string",
+                                                "default": "false"
+                                            }
+                                        }
+                                    }
+                                },
+                                "default": {
+                                    "ja": {
```

Comment:
I've put `"required": ["ja"]` now. Not sure if this impacts importing settings or anything like that?

Reply:
I don't know if that's needed, I'm just thinking about how this property is defined and how that affects things. It looks like the "translation" settings will only be shown if the language has those transforms specified, and based on this JSON schema, these are configurable. But there doesn't seem to be any way to configure which ones are available via the UI. That is, adding and removing them.

Is the expectation that if an entry isn't present for a language, e.g. for "en", it will use defaults? Because according to the spec, it's not required that anything be present, even Japanese. So could you potentially import a weird settings file and break stuff?

Reply:
In `languages-controller.js` there is 

```js
 async _updateOptions() {
        const options = await this._settingsController.getOptions();
        const {general: {language}} = options;
        if (!options.languages[language]) {
            await this._settingsController.modifyProfileSettings([{
                action: 'set',
                path: 'languages',
                value: {
                    ...options.languages,
                    [language]: {
                        textTransformations: {}
                    }
                }
            }]);
        }
    }
```
which doesn't look nice to me but makes sure that (if extension is used normally) `options[language].textTransformations` is defined, whose properties can then be set. Each textTransformation seems to be getting set to the default `'false'`.

Importing a settings file with e.g.
```json
"languages": {
    "en": {}
}
``` 
would, I think, knock things out of order, because `textTransformations` on `en` is missing, and the settings don't work on properties of missing objects. 

Setting `textTransformations` to a required property of each language should prevent such settings from being imported.

Reply:
Something else to consider here is that using dictionaries to store a dynamic collection of objects can have annoying edge use cases in the future. As an example, the "dictionaries" array used to be an object, but this made it very difficult to do things related to ordering, since while objects _do_ have an explicit order to the fields, it is not easy to adjust them.

I'll give more thoughts on this as the PR becomes more simple after rebasing the other sub-PRs I asked for, but it may be better to use arrays for both the language entries and their text transformations.

Something else to think about is what happens with the settings when we add a new transform in the future? I suppose settings upgrades can handle this, but food for thought nevertheless.

Reply:
I think the current setup allows adding textTransformations (and non-destructive changes to them) without settings upgrade functions, as requiring them might get out of hand with a large number of languages.

Reply:
I don't see anything analogous to dictionary order that might apply to these, though they can also be an array i guess, if that doesn't complicate thing too much.